### PR TITLE
mirror: add bucket exclusions

### PR DESCRIPTION
**Pull request by dormanze**
add bucket exclusions

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/mirror-main.go [hunk 1]
```diff
@@ -192,26 +196,30 @@ EXAMPLES:
       Exclude all .* files and *.temp files when mirroring.
       {{.Prompt}} {{.HelpName}} --exclude ".*" --exclude "*.temp" s3/test ~/test
 
-  10. Mirror objects newer than 10 days from bucket test to a local folder.
+  10. Mirror all buckets from aliased Amazon S3 cloud storage to a local folder.
+      Exclude test* buckets and backup* buckets when mirroring.
+      {{.Prompt}} {{.HelpName}} --exclude 'test*' --exclude 'backup*' s3 ~/test
```
Comment (shtripat) [inline cmd/mirror-main.go:201]: Option name is `--exclude-bucket`

File: cmd/mirror-url.go [hunk 2]
```diff
@@ -153,17 +173,27 @@ func deltaSourceTarget(ctx context.Context, sourceURL, targetURL string, opts mi
 		}
 
 		srcSuffix := strings.TrimPrefix(diffMsg.FirstURL, sourceURL)
-		// Skip the source object if it matches the Exclude options provided
+		// Skip the source objects if it matches the Exclude options provided
```
Comment (shtripat) [inline cmd/mirror-url.go:176]: its one object skipped at a time, so comment not needed to change I feel

File: cmd/mirror-url.go [hunk 3]
```diff
@@ -153,17 +173,27 @@ func deltaSourceTarget(ctx context.Context, sourceURL, targetURL string, opts mi
...
 		if matchExcludeOptions(opts.excludeOptions, srcSuffix, newClientURL(sourceURL).Type) {
 			continue
 		}
 
+		// Skip the source buckets if it matches the Exclude options provided
```
Comment (shtripat) [inline cmd/mirror-url.go:181]:
```suggestion
		// Skip the source bucket if it matches the Exclude options provided
```

File: cmd/mirror-url.go [hunk 4]
```diff
@@ -153,17 +173,27 @@ func deltaSourceTarget(ctx context.Context, sourceURL, targetURL string, opts mi
...
+		if matchExcludeBucketOptions(opts.excludeBuckets, srcSuffix) {
+			continue
+		}
+
 		tgtSuffix := strings.TrimPrefix(diffMsg.SecondURL, targetURL)
-		// Skip the target object if it matches the Exclude options provided
+		// Skip the target objects if it matches the Exclude options provided
```
Comment (shtripat) [inline cmd/mirror-url.go:187]:
```suggestion
		// Skip the target object if it matches the Exclude options provided
```

File: cmd/mirror-url.go [hunk 5]
```diff
@@ -153,17 +173,27 @@ func deltaSourceTarget(ctx context.Context, sourceURL, targetURL string, opts mi
...
 		if matchExcludeOptions(opts.excludeOptions, tgtSuffix, newClientURL(targetURL).Type) {
 			continue
 		}
 
+		// Skip the target buckets if it matches the Exclude options provided
```
Comment (shtripat) [inline cmd/mirror-url.go:192]:
```suggestion
		// Skip the target bucket if it matches the Exclude options provided
```

**Review by shtripat — APPROVED** [review]
lgtm

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**